source/solvers/navier_stokes_base.cc

```diff
@@ -1720,9 +1736,31 @@ void
 NavierStokesBase<dim, VectorType, DofsType>::read_checkpoint()
 {
   TimerOutput::Scope timer(this->computing_timer, "Read checkpoint");
-  std::string        prefix =
-    this->simulation_parameters.simulation_control.output_folder +
-    this->simulation_parameters.restart_parameters.filename;
+
+  // When the average velocity profile is used as an initial condition, the
+  // output folder path is not defined in the simulation control field but in
+  // the initial condition field.
+  std::string checkpoint_folder;
+  std::string checkpoint_file_name;
+  if (this->simulation_parameters.initial_condition->type ==
+        Parameters::InitialConditionType::average_velocity_profile &&
+      !(this->simulation_parameters.restart_parameters.restart))
+    {
+      checkpoint_folder =
+        this->simulation_parameters.initial_condition->checkpoint_folder;
+      checkpoint_file_name =
+        this->simulation_parameters.initial_condition->checkpoint_file_name;
+    }
+  else
+    {
+      checkpoint_folder =
+        this->simulation_parameters.simulation_control.output_folder;
+      checkpoint_file_name =
+        this->simulation_parameters.restart_parameters.filename;
+    }
```

Comment:
Wait what?
Here you don't mean the output path but the checkpoint path for the average_velocity right?

Reply:
I'm also a bit confused here, but I think it's due to variable names. As suggested by @blaisb, it could be worth renaming `checkpoint_folder `and `checkpoint_file_name`